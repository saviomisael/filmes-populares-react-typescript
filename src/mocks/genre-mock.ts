export const genreMock = [{ id: 1, name: 'Action' }];
