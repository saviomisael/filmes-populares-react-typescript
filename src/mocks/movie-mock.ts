export const movieMock = {
  id: 1,
  name: 'Duna',
  rating: 9,
  releaseDate: new Date(),
  poster: '/s9E9W77HS8zEQvsrpz5aEUTKnvD.jpg',
  overview:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
};
