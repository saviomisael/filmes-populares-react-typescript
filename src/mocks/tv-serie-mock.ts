export const tvSerieMock = { id: 1, name: 'Round 6', rating: 7.5, poster: '/6gcHdboppvplmBWxvROc96NJnmm.jpg' };
