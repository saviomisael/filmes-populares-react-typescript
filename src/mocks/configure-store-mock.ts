import { configureStore } from '@reduxjs/toolkit';
import { tvSerieMockReducer } from './tv-serie-duck-mock';

export const storeMock = configureStore({
  reducer: {
    tvSeries: tvSerieMockReducer,
  },
});
