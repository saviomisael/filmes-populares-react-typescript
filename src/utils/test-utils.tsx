import { render } from '@testing-library/react';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import { BaseStyles } from 'styles/base';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { storeMock } from 'mocks/configure-store-mock';

type optionsWithRouter = {
  initialRoutes?: string[];
  pathname: string;
};

export const renderTheme = (children: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <BaseStyles />
      {children}
    </ThemeProvider>,
  );

export const renderWithStoreProvider = (children: ReactNode) =>
  render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <BaseStyles />
        {children}
      </ThemeProvider>
    </Provider>,
  );

export const renderWithRouter = (children: ReactNode, options: optionsWithRouter) => {
  const initialRoutes = options.initialRoutes && options.initialRoutes.length > 0 ? options.initialRoutes : ['/'];

  return render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <MemoryRouter initialEntries={initialRoutes}>
          <Switch>
            <Route path={options.pathname} render={() => children} />
          </Switch>
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  );
};
