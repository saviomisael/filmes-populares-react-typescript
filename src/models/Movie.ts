import { IMAGE_URL } from 'utils/constants';
import { Genre } from './Genre';

export class Movie {
  constructor(
    public readonly id: number,
    public readonly name: string,
    public readonly rating: number,
    public readonly releaseDate: Date,
    public readonly poster: string,
    public readonly overview: string,
    public readonly genres?: Genre[],
  ) {
    this.poster = `${IMAGE_URL}${poster}`;
    this.genres = genres ?? [];
  }
}
