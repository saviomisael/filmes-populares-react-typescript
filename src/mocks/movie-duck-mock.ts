import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Movie } from 'models/Movie';
import { MovieState } from 'store/ducks/movie-duck';
import { genresMock } from './genre-mock';
import { movieListMock, movieMock } from './movie-mock';

const initialState: MovieState = {
  movies: movieListMock.map((x) => new Movie(x.id, x.name, x.rating, x.releaseDate, x.poster, x.overview)),
  singleMovie: new Movie(
    movieMock.id,
    movieMock.name,
    movieMock.rating,
    movieMock.releaseDate,
    movieMock.poster,
    movieMock.overview,
    [...genresMock],
  ),
};

const movieSlice = createSlice({
  name: 'movie',
  initialState,
  reducers: {
    addMovies: (state, action: PayloadAction<Movie[]>) => {
      state.movies = action.payload;
    },
    updateSingleMovie: (state, action: PayloadAction<Movie>) => {
      state.singleMovie = action.payload;
    },
  },
});

export const { actions: movieMockActions, reducer: movieMockReducer } = movieSlice;
