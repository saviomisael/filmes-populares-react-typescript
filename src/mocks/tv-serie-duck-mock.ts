import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { TvSerie } from 'models/TvSerie';
import { tvSerieListMock } from 'mocks/tv-serie-mock';

export interface TvSerieState {
  tvSeries: TvSerie[];
}

const initialState: TvSerieState = {
  tvSeries: tvSerieListMock.map((x) => new TvSerie(x.id, x.name, x.rating, x.poster)),
};

const tvSerieSlice = createSlice({
  name: 'tvserie',
  initialState,
  reducers: {
    addSeries: (state, action: PayloadAction<TvSerie[]>) => {
      state.tvSeries = action.payload;
    },
  },
});

export const { actions: tvSerieMockActions, reducer: tvSerieMockReducer } = tvSerieSlice;
