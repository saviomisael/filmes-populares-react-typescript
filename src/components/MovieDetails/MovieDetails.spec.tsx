import { screen } from '@testing-library/react';
import { movieMock } from 'mocks/movie-mock';
import { Movie } from 'models/Movie';
import { renderTheme } from 'utils/test-utils';
import { MovieDetails } from '.';

describe('<MovieDetails />', () => {
  it('should renders release date with custom text', () => {
    const movie = new Movie(
      movieMock.id,
      movieMock.name,
      movieMock.rating,
      movieMock.releaseDate,
      movieMock.poster,
      movieMock.overview,
    );

    renderTheme(<MovieDetails movie={movie} />);

    const releaseDateText = screen.getByText(/^lanÃ§ou em:(.)+2021$/i);

    expect(releaseDateText).toBeInTheDocument();
  });

  it('should renders alt text with custom text', () => {
    const movie = new Movie(
      movieMock.id,
      movieMock.name,
      movieMock.rating,
      movieMock.releaseDate,
      movieMock.poster,
      movieMock.overview,
    );

    renderTheme(<MovieDetails movie={movie} />);

    const altText = screen.getByAltText(/^poster do filme Duna$/i);

    expect(altText).toBeInTheDocument();
  });
});
