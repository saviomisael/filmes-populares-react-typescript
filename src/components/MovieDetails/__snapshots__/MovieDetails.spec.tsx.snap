// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetails /> should matches with snapshot 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  border-radius: 0.6rem;
  padding: 1rem 2rem;
  text-align: center;
  font-weight: 400;
  border: 0.2rem solid #FA9F42;
  font-size: 1.8rem;
  color: #FFFFFF;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  color: #FFFFFF;
  text-align: center;
  text-transform: none;
  padding-bottom: 2rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-column-gap: 1.4rem;
  column-gap: 1.4rem;
  color: #FFFFFF;
}

.c7::before {
  content: url(star.svg);
  -webkit-filter: invert(72%) sepia(86%) saturate(526%) hue-rotate(8deg) brightness(170%) contrast(101%);
  filter: invert(72%) sepia(86%) saturate(526%) hue-rotate(8deg) brightness(170%) contrast(101%);
}

.c6 {
  font-weight: 400;
  font-size: 1.8rem;
  color: #FFFFFF;
  text-align: center;
}

.c0 {
  margin: 0 auto;
  width: 70vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 3rem 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 5rem;
  row-gap: 1.3rem;
}

.c8 {
  width: 50%;
  object-fit: cover;
}

<div
  class="c0"
>
  <aside
    class="c1"
  >
    <h2
      class="c2"
    >
      Duna
    </h2>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Action
        </p>
      </div>
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Adventure
        </p>
      </div>
    </div>
    <p
      class="c6"
    >
      Lan√ßou em: 15 de fevereiro de 2021
    </p>
    <h4
      class="c7"
    >
      9
    </h4>
    <p
      class="c6"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </p>
  </aside>
  <img
    alt="poster do filme Duna"
    class="c8"
    src="https://image.tmdb.org/t/p/original/s9E9W77HS8zEQvsrpz5aEUTKnvD.jpg"
  />
</div>
`;
